src/Language/Spectacle/Syntax/Modal.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiWayIf #-}
+{-# LANGUAGE TupleSections #-}
+
+module Language.Spectacle.Syntax.Modal
+  ( Modal (Modal),
+    Effect (Always, UpUntil),
+    always,
+    eventually,
+    upUntil,
+    testModalResult1,
+  )
+where
+
+import Control.Monad
+import Data.Coerce
+import Data.Void
+
+import Data.Functor.Loom
+import Language.Spectacle.Lang
+import Language.Spectacle.Syntax.Modal.Internal
+import Language.Spectacle.Syntax.Modal.Quoted
+import Language.Spectacle.Syntax.Logic
+import Language.Spectacle.Syntax.NonDet
+import Language.Spectacle.Syntax.Error
+import Language.Spectacle.Exception.RuntimeException
+import Data.Function
+
+-- -------------------------------------------------------------------------------------------------
+
+always :: Member Modal effs => Lang ctx effs Bool -> Lang ctx effs Bool
+always m = scope (Always m)
+
+eventually :: Member Modal effs => Lang ctx effs Bool -> Lang ctx effs Bool
+eventually = upUntil (pure True)
+
+upUntil :: Member Modal effs => Lang ctx effs Bool -> Lang ctx effs Bool -> Lang ctx effs Bool
+upUntil m n = scope (UpUntil m n)
+
+
+runModal ::
```

Comment:
This is a bit of a nitpick since it's somewhat obvious that `runModal` is for running the Modal effect but for completeness sake, adding a documentation line might be nice for generated haddocks.